deepinv/physics/functional/random.py

```diff
@@ -0,0 +1,70 @@
+from typing import Tuple
+from warnings import warn
+
+import numpy as np
+import torch
+
+
+def random_choice(
```

Comment:
I know it's a bit laborious, but please could you add a test for this function? Maybe just a hardcoded test to test the rng (i.e. `random_choice(...) == torch.Tensor([...])`, and something to test the replacement and p, like:
```python
a = torch.Tensor([1,2,3,4,5])
p = torch.Tensor([0,0,1,0,0])
random_choice(a, 2, replace=True, p=p) 
```
should equal to [3,3].

Reply:
But thanks for providing this implementation! Is it taken from anywhere?

Reply:
I'll try for the tests. I implemented from scratch with the same logic and error as numpy, there is some examples provided in numpy docs for the case with equal probability. I'm not sure if there exists a better implementation out there.